IndexOutOfRangeException when specifying simple/complex input out-of-bounds

This is just an opportunity for a better error message.

This shader, which I accidentally specified an input count of 0 for:
```cs
[D2DInputCount(0)]
[D2DInputSimple(0)]
[D2DInputDescription(0, D2D1Filter.MinMagMipPoint)]
private readonly partial struct MultiplyBy255Shader
    : ID2D1PixelShader
{
    public float4 Execute()
    {
        float4 color = D2D.GetInput(0);
        return color * 255.0f;
    }
}
```

Produces this error output at compile time. Definitely could just say "hey dummy, you said 0 inputs, why're you specifying an input?" 😎

```
34>CSC : warning CS8785: Generator 'ID2D1ShaderGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'IndexOutOfRangeException' with message 'Index was outside the bounds of the array.'
34>D:\src\pdn\src\Framework\Direct2D1\Effects\ConvertToClassicPdnFormatEffect.cs(67,11,67,44): error CS0535: 'ConvertToClassicPdnFormatEffect.MultiplyBy255Shader' does not implement interface member 'ID2D1Shader.InitializeFromDispatchData(ReadOnlySpan<byte>)'
34>D:\src\pdn\src\Framework\Direct2D1\Effects\ConvertToClassicPdnFormatEffect.cs(67,11,67,44): error CS0535: 'ConvertToClassicPdnFormatEffect.MultiplyBy255Shader' does not implement interface member 'ID2D1Shader.GetPixelOptions()'
34>D:\src\pdn\src\Framework\Direct2D1\Effects\ConvertToClassicPdnFormatEffect.cs(67,11,67,44): error CS0535: 'ConvertToClassicPdnFormatEffect.MultiplyBy255Shader' does not implement interface member 'ID2D1Shader.GetInputCount()'
34>D:\src\pdn\src\Framework\Direct2D1\Effects\ConvertToClassicPdnFormatEffect.cs(67,11,67,44): error CS0535: 'ConvertToClassicPdnFormatEffect.MultiplyBy255Shader' does not implement interface member 'ID2D1Shader.GetInputType(uint)'
34>D:\src\pdn\src\Framework\Direct2D1\Effects\ConvertToClassicPdnFormatEffect.cs(67,11,67,44): error CS0535: 'ConvertToClassicPdnFormatEffect.MultiplyBy255Shader' does not implement interface member 'ID2D1Shader.LoadInputDescriptions<TLoader>(ref TLoader)'
34>D:\src\pdn\src\Framework\Direct2D1\Effects\ConvertToClassicPdnFormatEffect.cs(67,11,67,44): error CS0535: 'ConvertToClassicPdnFormatEffect.MultiplyBy255Shader' does not implement interface member 'ID2D1Shader.GetOutputBuffer(out uint, out uint)'
34>D:\src\pdn\src\Framework\Direct2D1\Effects\ConvertToClassicPdnFormatEffect.cs(67,11,67,44): error CS0535: 'ConvertToClassicPdnFormatEffect.MultiplyBy255Shader' does not implement interface member 'ID2D1Shader.LoadDispatchData<TLoader>(ref TLoader)'
34>D:\src\pdn\src\Framework\Direct2D1\Effects\ConvertToClassicPdnFormatEffect.cs(67,11,67,44): error CS0535: 'ConvertToClassicPdnFormatEffect.MultiplyBy255Shader' does not implement interface member 'ID2D1Shader.BuildHlslSource(out string)'
34>D:\src\pdn\src\Framework\Direct2D1\Effects\ConvertToClassicPdnFormatEffect.cs(67,11,67,44): error CS0535: 'ConvertToClassicPdnFormatEffect.MultiplyBy255Shader' does not implement interface member 'ID2D1Shader.LoadBytecode<TLoader>(ref TLoader, D2D1ShaderProfile?, D2D1CompileOptions?)'
```

--- a/src/ComputeSharp.D2D1.SourceGenerators/ID2D1ShaderGenerator.CreateLoadBytecodeMethod.cs
+++ b/src/ComputeSharp.D2D1.SourceGenerators/ID2D1ShaderGenerator.CreateLoadBytecodeMethod.cs
@@ -116,6 +116,12 @@ partial class ID2D1ShaderGenerator
         /// <returns>Whether the shader only has simple inputs.</returns>
         public static bool IsSimpleInputShader(INamedTypeSymbol structDeclarationSymbol, int inputCount)
         {
+            // If there are no inputs, the shader is as if only had simple inputs
+            if (inputCount == 0)
+            {
+                return true;
+            }
+
             // Build a map of all simple inputs (unmarked inputs default to being complex)
             bool[] simpleInputsMap = new bool[inputCount];
 
@@ -123,8 +129,11 @@ partial class ID2D1ShaderGenerator
             {
                 switch (attributeData.AttributeClass?.GetFullMetadataName())
                 {
-                    case "ComputeSharp.D2D1.D2DInputSimpleAttribute":
-                        simpleInputsMap[(int)attributeData.ConstructorArguments[0].Value!] = true;
+                    // Only retrieve indices of simple inputs that are in range. If an input is out of
+                    // range, the diagnostic for it will already be emitted by a previous generator step.
+                    case "ComputeSharp.D2D1.D2DInputSimpleAttribute"
+                    when attributeData.ConstructorArguments[0].Value is int index && index < inputCount:
+                        simpleInputsMap[index] = true;
                         break;
                 }
             }

